src/restd.c

```diff
@@ -35,11 +35,9 @@
 */
```

Comment:
What I really don't get is why some parts of the library are in C, and other parts in Fortran. How is the decision made as to what language to use for what?

In other libraries, C forms the lowest level of code, and is called by Fortran, but I've never seen the opposite anywhere else except NCEPLIBS-bufr. That is, I don't know of any other libraries where C code calls Fortran code.

Reply:
For better or worse, this is just how the library evolved over many years/decades.  In some cases (e.g. the logic in restd.c and stseq.c) we needed to use recursion, which Fortran77 didn't universally support on all platforms at the time.  Similarly, not all platforms supported the reading and writing of binary BUFR data in Fortran without the use of kludgy extensions or blocking mechanisms, and older versions of Fortran didn't always support things like structures, or intrinsic functions such as qsort and bsearch which were needed for sorting and searching through large BUFR master tables.  But since all of this functionality was readily available in standard C, and since there was a universally-supportable way to link Fortran to C (albeit using the aforementioned UNDERSCORE and other "shenanigans" ;-), then the decision was made to use C where and when it was needed in the library.

I get that more modern versions of Fortran (95, 2003), etc. can now pretty-much do everything that C can do, and as you know we've been taking advantages of these modern features as we continue to maintain the library, but at this point I'm not keen to go back and rewrite all of the existing C functionality in Fortran just for consistency with other libraries.